arm64: Align clear_page() to 16 bytes

With automatic zeroing of pages upon allocation enabled, time spent in
clear_page() is significant according to perf top. Since clear_page() is
used in loops, align it to 16 bytes, which significantly reduces CPU time
spent in clear_page()

Signed-off-by: Sultan Alsawaf <sultan@kerneltoast.com>
---
 arch/arm64/lib/clear_page.S | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/lib/clear_page.S b/arch/arm64/lib/clear_page.S
index 1fd5d790ab80..931dd4be6593 100644
--- a/arch/arm64/lib/clear_page.S
+++ b/arch/arm64/lib/clear_page.S
@@ -14,6 +14,7 @@
  * Parameters:
  *	x0 - dest
  */
+	.p2align 4
 SYM_FUNC_START(__pi_clear_page)
 	mrs	x1, dczid_el0
 	tbnz	x1, #4, 2f	/* Branch if DC ZVA is prohibited */

